scripts/config.py

```diff
@@ -348,21 +348,25 @@ def _config_compiler(self):
         if self.configDict["OPENCV"] == "" or self.configDict["OPENCVSUPPORTSCUDA"] or self.configDict["OPENCV3"]:
             self._config_OpenCV()
 
-    def _get_GCC_version(self, compiler):
+
+    def _get_gcc_version(self, gcc):
         log = []
-        runJob(compiler + " -dumpversion", show_output=False,
+        runJob(gcc + " -v", show_output=False,
                show_command=False, log=log)
-        full_version = log[0].strip()
+        # find 'gcc version' line (last one)
+        # expected format: '...\n gcc version X.X.X (... X.X.X.X-X...)'
+        full_version_line = next(l for l in log if 'gcc version ' in l)
```

Comment:
`in l.lower()`?
Also, we had something like that before, but we went for `-dumpversion` because it was more reliable. Any reason to change it back?

Reply:
-dumpversion return just the major version (at least in my ubuntu/gcc and [others](https://stackoverflow.com/questions/45168516/gcc-7-1-1-on-fedora-26-dumpversion-now-only-includes-major-version-by-default)), I have recovered the previous function and added -dumpfullversion call if -dumpversion returns just one digit.